matchzoo/pack.py

```diff
@@ -0,0 +1,52 @@
+"""Convert list of input into class:`DataPack` expected format."""
+
+import pandas as pd
+
+from matchzoo import datapack
+
+
+def infer_stage(data: list):
+    if data and len(data[0]) == 5:
+        return 'train'
```

Comment:
Here, stage `evaluate` must be kept. Both `train` and `evaluate` need the `label`. While in preprocess. Only `train` can call `fit_transform`, and `evaluate` and `predict` should only call `transform`.